### DISCUSSION TITLE
nvm use exit status 1 even after following all intructions

### DISCUSSION THREAD

**Discussion by jhp829** [coreybutler/nvm-windows#996] [Support]
**Issue:**
exit status 1 when using nvm use. I have followed the [common issues page](https://github.com/coreybutler/nvm-windows/wiki/Common-Issues#permissions-exit-1-exit-5-access-denied-exit-145) to try to solve the issue and nothing worked.

**How To Reproduce:**

Turn on developer mode
Uninstall any previous versions of NVM
Install newest version of NVM
Set install paths to a new folder made in C:/ 
Open powershell as administrator
nvm installs lts (18.16.1)
nvm use lts

**Expected behavior:**
Node version 18.16.1 should be enabled

**Observed Behavior:**

Duplicate exit status 1 messages. The exit status is not followed by any message.

exit status 1:
exit status 1:

**Things tried"**

I have tried all of these combined:

- Creating new folders in various locations
- Turning on developer mode
- Running powershell as administrator
- Running nvm installer as administrator
- Uninstalling any previous versions of node

**Desktop:**

OS: Windows 11 latest

**Comment by coreybutler** [comment]
What are the results of `nvm debug`?

**Reply by jhp829** [reply]
Currently I get this:

`Running NVM for Windows with administrator privileges.

Administrator: Windows PowerShell
NVM4W Version:      1.1.11
NVM4W Path:         C:\Users\jbae0\AppData\Roaming\node\nvm\nvm.exe
NVM4W Settings:     C:\Users\jbae0\AppData\Roaming\node\nvm\settings.txt
NVM_HOME:           C:\Users\jbae0\AppData\Roaming\node\nvm
NVM_SYMLINK:        C:\Users\jbae0\AppData\Roaming\node
Node Installations: C:\Users\jbae0\AppData\Roaming\node\nvm

Active Node.js Version: none
(run "nvm use <version>" to activate a version)

No problems detected.

Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues`

**Reply by coreybutler** [reply]
Is `nvm install lts` throwing the `exit 1`, or is it `nvm use lts`? Check `C:\Users\jbae0\AppData\Roaming\node\nvm` to see if there is a directory call `v18.16.1` and whether it contains any files.

It's very unusual to see no error message because NVM4W outputs its error messages. Is Windows Defender (or other antivirus) running? Is Controlled folder access enabled (Windows Security > Virus & threat protection > Controlled folder access)? If Windows silently denies any of the basic operations (such as saving/extracting/copying the Node download) NVM4W would not receive an error message. In this case, the fix would be allow-listing NV4W directories (`NVM_HOME` & `NVM_SYMLINK`).

**Reply by jhp829** [reply]
`nvm use lts` is throwing these exit statuses. 

Inside the nvm file, there is a directory `v18.16.1` and it has files that appear to be the correct install. I also have controlled folder access disabled on the settings.

**Reply by coreybutler** [reply]
If you have any other antivirus installed, you may want to _temporarily_ disable it. It's very unlikely to be the cause unless you're using a relatively unknown antivirus.

Did you try setting the version directly, i.e. `nvm use 18.16.1`?

You can also try running `C:\Users\jbae0\AppData\Roaming\node\nvm\v18.16.1\node.exe -v`. If this displays the correct version, then the issue is the symlink. If that is the case, I'd recommend installing a different version of Node (`nvm install ...`), `nvm use` the new version, then `nvm use lts` again. This would reset the symlink if it is corrupted.

The only other thing I can think of is to try uninstalling and reinstalling. Several users have succeeded with this in the past.

**Reply by jhp829** [reply]
I have disabled all antiviruses including everything in the Windows settings. I tried setting the version directly and also tried installing a different version of Node after which I got the same error. I also tried running `C:\Users\jbae0\AppData\Roaming\node\nvm\v18.16.1\node.exe -v` which showed the correct version. 

I tried uninstalling and reinstalling multiple times but I seem to get the same error every time.

**Reply by khuezy** [reply]
I'm on windows 10 and experiencing the same issue. I followed your recommendation of `nvm install latest` and the output says installation was complete but doing `nvm use <version>` outputs `exit status 1:` twice in a row.

**Reply by coreybutler** [reply]
@jhp829 You can clearly install Node, but the symlink is not being updated. One major reason for this is if there is a prior version of Node installed or some other physical directory where the `NVM_SYMLINK` should be. In your case, it would be `C:\Users\jbae0\AppData\Roaming\node`. You could delete this directory manually, allowing the symlink to be recreated the next time you run `nvm  use 18.16.1`.

**Reply by GitSTho** [reply]
I had the same problem, fixed it with administrative privileges as mentioned in the doc.
